core/src/main/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriter.java

```diff
@@ -144,7 +144,7 @@ public UnsafeShuffleWriter(
     this.sparkConf = sparkConf;
     this.transferToEnabled = sparkConf.getBoolean("spark.file.transferTo", true);
     this.initialSortBufferSize =
-      (int) sparkConf.get(package$.MODULE$.SHUFFLE_SORT_INIT_BUFFER_SIZE());
+      (int) (long) sparkConf.get(package$.MODULE$.SHUFFLE_SORT_INIT_BUFFER_SIZE());
```

Comment:
what if we dont' have this long cast?

Reply:
If we don't convert to long first , it will encounter exception like this:
`Caused by: java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer`

Reply:
how about `sparkConf.get(...).intValue`?